sdk/java/pulumi/src/main/java/com/pulumi/Pulumi.java

```diff
@@ -34,9 +35,45 @@ static void run(Consumer<Context> stack) {
      * @param stack the stack to run in Pulumi runtime
      * @return a future exit code from Pulumi runtime after running the stack
      * @see #run(Consumer)
+     * @see #withOptions(StackOptions)
      */
     static CompletableFuture<Integer> runAsync(Consumer<Context> stack) {
-        var pulumi = PulumiInternal.fromEnvironment();
-        return pulumi.runAsync(stack);
+        return withOptions(StackOptions.Empty).runAsync(stack);
+    }
+
+    /**
+     * @param options the {@link StackOptions} to use
+     * @return a Pulumi program entrypoint with given {@link StackOptions}
+     * @see #run(Consumer)
+     * @see #runAsync(Consumer)
+     */
+    static Pulumi.API withOptions(StackOptions options) {
```

Comment:
yes, I thought it would be more scalable for future changes as opposed to having a method just for transformations, but if it is controversial I'm not attached to the idea of `StackOptions` and I'm open to alternatives